drivers/storage/openstack/tests/openstack_test.go

```diff
@@ -0,0 +1,534 @@
+// +build !libstorage_storage_driver libstorage_storage_driver_openstack
+
+package openstack
+
+import (
+	"bytes"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	log "github.com/Sirupsen/logrus"
+	gofig "github.com/akutz/gofig/types"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/codedellemc/libstorage/api/context"
+	"github.com/codedellemc/libstorage/api/registry"
+	"github.com/codedellemc/libstorage/api/server"
+	apitests "github.com/codedellemc/libstorage/api/tests"
+	"github.com/codedellemc/libstorage/api/types"
+	"github.com/codedellemc/libstorage/api/utils"
+
+	// load the driver
+	openstack "github.com/codedellemc/libstorage/drivers/storage/openstack"
+	openstackx "github.com/codedellemc/libstorage/drivers/storage/openstack/executor"
+)
+
+// configYAML is an embedded config file that specifies the bare minimum
+// configuration for creating an openstack volume.
+var configYAML = []byte(`
+openstack:
+  authURL: ` + os.Getenv("OS_AUTH_URL") + `
+  username: ` + os.Getenv("OS_USERNAME") + `
+  tenantName: ` + os.Getenv("OS_TENANT_NAME") + `
+  domainName: ` + os.Getenv("OS_USER_DOMAIN_NAME") + `
+  password: ` + os.Getenv("OS_PASSWORD") + `
+  availabilityZoneName: ` + os.Getenv("OS_AVAILABILITY_ZONE") + `
+  regionName: ` + os.Getenv("OS_REGION_NAME") + `
```

Comment:
The calls to `os.Getenv` are not necessary. The test will already pull in any matching env vars, just the same as if you were running libStorage/REX-Ray normally. For example, the EBS driver does not pull any of those options in: https://github.com/codedellemc/libstorage/blob/master/drivers/storage/ebs/tests/ebs_test.go#L34-L37

But if you look at the README in the test directory, the examples show using the env vars.